ios/RNPusherPushNotifications.m

```diff
@@ -21,45 +22,46 @@ - (dispatch_queue_t)methodQueue
 {
     dispatch_async(dispatch_get_main_queue(), ^{
         RCTLogInfo(@"Creating pusher with App Key: %@", appKey);
-        // Pusher init
-        self.pusher = [PTPusher pusherWithKey:appKey delegate:self encrypted:YES];
-        UIUserNotificationType notificationTypes = UIUserNotificationTypeAlert | UIUserNotificationTypeBadge | UIUserNotificationTypeSound;
-        UIUserNotificationSettings *pushNotificationSettings = [UIUserNotificationSettings settingsForTypes:notificationTypes categories: NULL];
-        [[UIApplication sharedApplication ] registerUserNotificationSettings:pushNotificationSettings];
-        [[UIApplication sharedApplication] registerForRemoteNotifications];
-
+        [[PushNotifications shared] startWithInstanceId:appKey];
+        [[PushNotifications shared] registerForRemoteNotifications];
     });
 }
 
 RCT_EXPORT_METHOD(subscribe:(NSString *)interest)
 {
     dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
         RCTLogInfo(@"Subscribing to: %@", interest);
-        [[[self pusher] nativePusher] subscribe:(NSString *)interest];
+        NSError *anyError;
+        [[PushNotifications shared] subscribeWithInterest:interest error:&anyError completion:^{
```

Comment:
It would be good if errors would be handled.
We could do something as simple as [this](https://github.com/pusher/push-notifications-swift/blob/master/push-notifications-objc/push-notifications-objc/AppDelegate.m#L21).